gdal/ogr/ogrfeature.cpp

```diff
@@ -2036,14 +2036,54 @@ static void OGRFeatureFormatDateTimeBuffer( char szTempBuffer[TEMP_BUFFER_SIZE],
 {
     const int ms = OGR_GET_MS(fSecond);
     if( ms != 0 )
-        CPLsnprintf( szTempBuffer, TEMP_BUFFER_SIZE,
-                  "%04d/%02d/%02d %02d:%02d:%06.3f",
-                  nYear,
-                  nMonth,
-                  nDay,
-                  nHour,
-                  nMinute,
-                  fSecond );
+    {
+        if ( ms == 1000 ) 
```

Comment:
This logic for dealing with ms == 1000 is repeated 3 times. It would be good to have it in a single function called from those 3 call sites